doc/nrf/protocols/matter/getting_started/matter_api.rst

```diff
@@ -0,0 +1,248 @@
+   #endif
+     /** @brief Custom code to execute in the Matter main event loop before the server initialization. */
+     CustomInit mPreServerInitClbk{ nullptr };
+     /** @brief Custom code to execute in the Matter main event loop after the server initialization. */
+     CustomInit mPostServerInitClbk{ nullptr };
+
+     /** @brief Default implementation static objects that will be stripped by the compiler when above
+      * pointers are overwritten by the application. */
+   #ifdef CONFIG_CHIP_WIFI
+     static chip::app::Clusters::NetworkCommissioning::Instance sWiFiCommissioningInstance;
+   #endif
+     static chip::CommonCaseDeviceServerInitParams sServerInitParamsDefault;
+   #ifdef CONFIG_CHIP_FACTORY_DATA
+     static chip::DeviceLayer::FactoryDataProvider<chip::DeviceLayer::InternalFlashFactoryData> sFactoryDataProviderDefault;
+   #endif
+   #ifdef CONFIG_CHIP_CRYPTO_PSA
+     static chip::Crypto::PSAOperationalKeystore sOperationalKeystoreDefault;
+   #endif
+   };
+
+In this structure, each Matter component is defined as a public pointer to its generic interface that can be set with the concrete object.
+Moreover, each pointer is initially assigned with the default concrete implementation of the given interface that is stripped by the compiler in case the user overwrites it with customized implementation in the application.
+
+The nRF Connect Matter API consists of the following functions that can be used to initialize Matter components in proper order:
+
+* :c:func:`PrepareServer()`:
+
+This function schedules initialization of Matter components including memory, server configuration and networking backend.
+Depending on the selected Kconfig options the initialization may also include factory data and operational key storage.
+Scheduling all initialization procedures to the Matter thread aims to provide a synchronization between all components and the application code.
+This function accepts :c:struct:`InitData` argument that contains implementation of all required Matter interfaces.
+If no argument is provided, this function uses the default-constructed :c:struct:`InitData` temporary object.
+After this function is used, the :c:func:`StartServer()` must be called to start the Matter thread, eventually execute the initialization and wait to synchronize the caller's thread with the Matter thread.
+
+* :c:func:`StartServer()`:
+
+This is a blocking function that starts the Matter thread and waits until all Matter server components are initialized.
+
+* :c:func:`GetFactoryDataProvider()`:
+
+This function returns generic pointer to the ``FactoryDataProvider`` object that was set during the initialization.
+It may come in handy when there is need to access factory data at the Matter server initialization stage or as a part of the post initialization callback (``mPostServerInitClbk`` in :c:struct:`InitData`).
+This function is available only if the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` Kconfig option is selected.
```

Comment:
```suggestion
:c:func:`GetFactoryDataProvider()`:
  This function returns the generic pointer to the ``FactoryDataProvider`` object that was set during the initialization.
  It can be used when you need to access factory data at the Matter server initialization stage or as a part of the post initialization callback (``mPostServerInitClbk`` in :c:struct:`InitData`).
  
  This function is only available if the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` Kconfig option is selected.
```